internal/adapters/forwarder/events_forwarder/events_forwarder.go

```diff
@@ -0,0 +1,164 @@
+// MIT License
+//
+// Copyright (c) 2021 TFG Co
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package events_forwarder
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/topfreegames/maestro/internal/core/ports/forwarder"
+
+	"github.com/topfreegames/maestro/internal/core/entities/events"
+	entities "github.com/topfreegames/maestro/internal/core/entities/forwarder"
+	"github.com/topfreegames/maestro/internal/core/ports/errors"
+
+	pb "github.com/topfreegames/protos/maestro/grpc/generated"
+)
+
+type eventsForwarder struct {
+	forwarderClient forwarder.ForwarderClient
+}
+
+func NewEventsForwarder(forwarderClient forwarder.ForwarderClient) *eventsForwarder {
+	return &eventsForwarder{
+		forwarderClient: forwarderClient,
+	}
+}
+
+// ForwardRoomEvent forwards room events. It receives the room event attributes and forwarder configuration.
+func (f *eventsForwarder) ForwardRoomEvent(ctx context.Context, eventAttributes events.RoomEventAttributes, forwarder entities.Forwarder) error {
+	switch eventAttributes.EventType {
+	case events.Arbitrary:
+		event := pb.RoomEvent{
+			Room: &pb.Room{
+				Game:     eventAttributes.Game,
+				RoomId:   eventAttributes.RoomId,
+				Host:     eventAttributes.Host,
+				Port:     eventAttributes.Port,
+				Metadata: *fromMapInterfaceToMapString(eventAttributes.Other),
+			},
+			EventType: events.FromRoomEventTypeToString(eventAttributes.EventType),
+			Metadata:  f.mergeInfos(forwarder.Options.Metadata, eventAttributes.Other),
```

Comment:
@lyracampos You can remove this Metadata field from RoomEvent request, we only need the room metadata (already defined above)